src/ansys/mapdl/core/math.py

```diff
@@ -686,13 +684,24 @@ def get_vec(self, dtype=np.double, fname="file.full", mat_id="RHS", asarray=Fals
             "Call MAPDL to extract the %s vector from the file %s", mat_id, fname
         )
 
+        if mat_id.upper() not in ["RHS", "GVEC", "BACK", "FORWARD"]:
+            raise ValueError(
+                f"The 'mat_id' value ({mat_id}) is not allowed."
+                'Only "RHS", "GVEC", "BACK", or "FORWARD" are allowed.'
+            )
+
+        if mat_id.upper() in ["BACK", "FORWARD"] and not dtype:
+            dtype = np.int32
+        else:
+            dtype = np.double
+
         fname = self._load_file(fname)
         self._mapdl.run(
             f"*VEC,{name},{MYCTYPE[dtype]},IMPORT,FULL,{fname},{mat_id}", mute=True
         )
         ans_vec = AnsVec(name, self._mapdl)
         if asarray:
-            return self._mapdl._vec_data(ans_vec.id)
+            return self._mapdl._vec_data(ans_vec.id).astype(dtype)
```

Comment:
With `copy=False` we avoid copying if possible.

```suggestion
            return self._mapdl._vec_data(ans_vec.id).astype(dtype, copy=False)
```